src/preferences/dialog/dlgprefsounditem.cpp

```diff
@@ -185,8 +184,8 @@ void DlgPrefSoundItem::loadPath(const SoundManagerConfig &config) {
  * config.
  */
 void DlgPrefSoundItem::writePath(SoundManagerConfig *config) const {
-    SoundDevice *device = getDevice();
-    if (device == NULL) {
+    QSharedPointer<SoundDevice> pDevice = getDevice();
+    if (pDevice.isNull()) {
```

Comment:
```
if (!pDevice) {
...
```
Use operator bool() instead of Qt-specific member functions like isNull() to keep the code compatible with std::shared_ptr
This is the lesson we learned when migrating TrackPointer from QSharedPointer to std::shared_ptr